File: app/src/main/java/com/example/xcards/domain/adapters/AdapterForRecyclerView.java
package com.example.xcards.domain.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import com.example.xcards.R;
import com.example.xcards.data.CardData;

import java.util.ArrayList;

public class AdapterForRecyclerView extends RecyclerView.Adapter<AdapterForRecyclerView.ViewHolder> {
    ArrayList<CardData> cardsArray;
    Context context;

    public AdapterForRecyclerView(Context context, ArrayList images) {
        this.context = context;
        this.cardsArray = images;
    }

    @NonNull
    @Override
    public AdapterForRecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        // Inflating the Layout(Instantiates list_item.xml layout file into View object)
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.card, parent, false);

        // Passing view to ViewHolder
        AdapterForRecyclerView.ViewHolder viewHolder = new AdapterForRecyclerView.ViewHolder((CardView) view);
        return viewHolder;
    }

    // Binding data to the into specified position
    @Override
    public void onBindViewHolder(@NonNull AdapterForRecyclerView.ViewHolder holder, int position) {
        // TypeCast Object to int type
        CardData displayingCard = cardsArray.get(position);
        holder.cardName.setText(displayingCard.getNameModule());
        holder.cardsCount.setText(displayingCard.getCardsCount());
    }

    @Override
    public int getItemCount() {
        // Returns number of items currently available in Adapter
        return cardsArray.size();
    }

    // Initializing the Views
    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView cardName;
        TextView cardsCount;

        public ViewHolder(CardView view) {
            super(view);
            cardName = view.findViewById(R.id.textViewName);
            cardsCount = view.findViewById(R.id.textViewCount);
        }
    }
}

File: app/src/main/java/com/example/xcards/domain/adapters/AdapterForRecyclerView.kt
package com.example.xcards.domain.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.xcards.R
import com.example.xcards.data.CardData

class AdapterForRecyclerView(var context: Context?, images: ArrayList<*>) :
    RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    var cardsArray: ArrayList<CardData>

    init {
        cardsArray = images as ArrayList<CardData>
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        if (viewType == 0) {
            val view = LayoutInflater.from(parent.context).inflate(R.layout.buttons, parent, false)
            return ButtonsViewHolder(view)
        } else {
            // Inflating the Layout(Instantiates list_item.xml layout file into View object)
            val view = LayoutInflater.from(parent.context).inflate(R.layout.card, parent, false)

            // Passing view to ViewHolder
            return ViewHolder(view)
        }
    }

    override fun getItemViewType(position: Int): Int {
        return if (position == 0) 0 else 1
    }

    // Binding data to the into specified position
    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        if (position - 1 > 0) {
            val (nameModule, cardsCount) = cardsArray[position - 1]
            (holder as ViewHolder).cardName.text = nameModule
            holder.cardsCount.text = cardsCount.toString()
        }
    }

    override fun getItemCount(): Int {
        // Returns number of items currently available in Adapter
        return cardsArray.size + 1
    }

    inner class ButtonsViewHolder(view: View?) : RecyclerView.ViewHolder(
        view!!
    )

    // Initializing the Views
    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var cardName: TextView
        var cardsCount: TextView

        init {
            cardName = view.findViewById(R.id.textViewName)
            cardsCount = view.findViewById(R.id.textViewCount)
        }
    }
}